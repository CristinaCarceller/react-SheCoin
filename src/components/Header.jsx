import "./styles/header.css";
import { Link, useNavigate } from "react-router-dom";
import { Toast } from "bootstrap";

const Header = () => {
	const accessToken = localStorage.getItem("accessToken");
	const navigate = useNavigate();

	const handleLogout = () => {
		localStorage.clear();
		const toastTrigger = document.getElementById("toastTrigger");
		const toast = new Toast(toastTrigger);
		toast.show();
		navigate("/home");
	};

	return (
		<>
			<div className="header sticky-top">
				<nav className="navbar navbar-expand-lg" id="navbar">
					<div className="container-fluid" id="container">
						<nav className="navbar">
							<div className="container">
								<div className="navbar-brand">
									<img
										src="./src/components/images/SheCoin.png"
										width={130}
										height={50}
										style={{ padding: "0px", margin: "0px" }}></img>
								</div>
							</div>
						</nav>
						<button
							className="navbar-toggler"
							type="button"
							data-bs-toggle="collapse"
							data-bs-target="#navbarNav"
							aria-controls="navbarNav"
							aria-expanded="false"
							aria-label="Toggle navigation">
							<span className="navbar-toggler-icon"></span>
						</button>
						<div className="collapse navbar-collapse" id="navbarNav">
							<ul className="navbar-nav">
								<li className="nav-item">
									<Link to="/home" className="nav-link" id="homeLink">
										Home
									</Link>
								</li>
								{accessToken && (
									<li className="nav-item">
										<Link
											to="/watchlist"
											className="nav-link"
											id="watchlistLink">
											Watchlist
										</Link>
									</li>
								)}
								<li className="nav-item">
									{!accessToken && (
										<button
											className="btn btn-primary"
											type="submit"
											id="login">
											<Link to="/login" id="loginLink">
												Login
											</Link>
										</button>
									)}
								</li>
								<li className="nav-item">
									{accessToken && (
										<button
											className="btn btn-primary me-2"
											type="submit"
											id="logout"
											onClick={handleLogout}>
											Logout
										</button>
									)}
								</li>
							</ul>
						</div>
					</div>
				</nav>
			</div>
			<div
				className="toast-container position-fixed top-0 end-0 p-3"
				id="toastContainer">
				<div
					className="toast"
					id="toastTrigger"
					role="alert"
					aria-live="assertive"
					aria-atomic="true"
					style={{ backgroundColor: "#7f7caf", color: "white" }}>
					<div
						className="toast-header"
						style={{ backgroundColor: "#7f7caf", color: "white" }}>
						<strong className="me-auto">Logout</strong>
						<button
							type="button"
							className="btn-close"
							data-bs-dismiss="toast"
							aria-label="Close"></button>
					</div>
					<div className="toast-body">
						You have been logged out successfully.
					</div>
				</div>
			</div>
		</>
	);
};

export default Header;
