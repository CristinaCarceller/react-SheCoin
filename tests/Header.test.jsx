import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom";
import userEvent from "@testing-library/user-event";
import { MemoryRouter } from "react-router-dom";
import Header from "../src/components/Header.jsx";
import { describe, it, expect } from "vitest";

describe("Header Component", () => {
	it("should match the snapshot", () => {
		const { asFragment } = render(
			<MemoryRouter>
				<Header />
			</MemoryRouter>
		);
		expect(asFragment()).toMatchSnapshot();
	});
});
