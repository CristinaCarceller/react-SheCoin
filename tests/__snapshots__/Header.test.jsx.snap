// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Header Component > should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="header sticky-top"
  >
    <nav
      class="navbar navbar-expand-lg"
      id="navbar"
    >
      <div
        class="container-fluid"
        id="container"
      >
        <nav
          class="navbar"
        >
          <div
            class="container"
          >
            <div
              class="navbar-brand"
            >
              <img
                height="50"
                src="./src/components/images/SheCoin.png"
                style="padding: 0px; margin: 0px;"
                width="130"
              />
            </div>
          </div>
        </nav>
        <button
          aria-controls="navbarNav"
          aria-expanded="false"
          aria-label="Toggle navigation"
          class="navbar-toggler"
          data-bs-target="#navbarNav"
          data-bs-toggle="collapse"
          type="button"
        >
          <span
            class="navbar-toggler-icon"
          />
        </button>
        <div
          class="collapse navbar-collapse"
          id="navbarNav"
        >
          <ul
            class="navbar-nav"
          >
            <li
              class="nav-item"
            >
              <a
                class="nav-link"
                href="/home"
                id="homeLink"
              >
                Home
              </a>
            </li>
            <li
              class="nav-item"
            >
              <button
                class="btn btn-primary"
                id="login"
                type="submit"
              >
                <a
                  href="/login"
                  id="loginLink"
                >
                  Login
                </a>
              </button>
            </li>
            <li
              class="nav-item"
            />
          </ul>
        </div>
      </div>
    </nav>
  </div>
  <div
    class="toast-container position-fixed top-0 end-0 p-3"
    id="toastContainer"
  >
    <div
      aria-atomic="true"
      aria-live="assertive"
      class="toast"
      id="toastTrigger"
      role="alert"
      style="background-color: rgb(127, 124, 175); color: white;"
    >
      <div
        class="toast-header"
        style="background-color: rgb(127, 124, 175); color: white;"
      >
        <strong
          class="me-auto"
        >
          Logout
        </strong>
        <button
          aria-label="Close"
          class="btn-close"
          data-bs-dismiss="toast"
          type="button"
        />
      </div>
      <div
        class="toast-body"
      >
        You have been logged out successfully.
      </div>
    </div>
  </div>
</DocumentFragment>
`;
